projects/RabbitMQ.Client/client/api/ConnectionFactory.cs

```diff
@@ -516,6 +612,30 @@ public IConnection CreateConnection(IList<AmqpTcpEndpoint> endpoints, string cli
             return CreateConnection(EndpointResolverFactory(endpoints), clientProvidedName);
         }
 
+        /// <summary>
+        /// Asynchronously create a connection using a list of endpoints. By default each endpoint will be tried
+        /// in a random order until a successful connection is found or the list is exhausted.
+        /// The selection behaviour can be overridden by configuring the EndpointResolverFactory.
+        /// </summary>
+        /// <param name="endpoints">
+        /// List of endpoints to use for the initial
+        /// connection and recovery.
+        /// </param>
+        /// <param name="clientProvidedName">
+        /// Application-specific connection name, will be displayed in the management UI
+        /// if RabbitMQ server supports it. This value doesn't have to be unique and cannot
+        /// be used as a connection identifier, e.g. in HTTP API requests.
+        /// This value is supposed to be human-readable.
+        /// </param>
+        /// <returns>Open connection</returns>
+        /// <exception cref="BrokerUnreachableException">
+        /// When no hostname was reachable.
+        /// </exception>
```

Comment:
Give this is the public API what is the benefit of passing IList VS IEnumerable or IReadOblyCollection?

Reply:
That's how the API has always been. I do have an item [here](https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1413) to review the use of collections, and I'll link back to your comment so it's not missed.